students/Osiddiquee/lesson04/trigram.py

```diff
@@ -0,0 +1,34 @@
+import random
+
+#reading in a text file
+inputfile = open('sherlock_small.txt')
+novel = inputfile.read()
+inputfile.close()
+
+# Builds the dicitonary with which the Kata is built
+def dictionary_builder(text):
+    words = text.replace('\n', ' ').split(' ')
```

Comment:
.split() with no arguments splits all whitespace including "\n" chars. Note that by replacing with " " means that you'll be putting a lot of whitespace in. Then since you're splitting by " " (one space) if  you have something like:
```
In [39]: print("these are words \n\n with \n spaces".replace("\n"," ").split(" "))
['these', 'are', 'words', '', '', '', 'with', '', '', 'spaces']

In [40]: print("these are words \n\n with \n spaces".split())
['these', 'are', 'words', 'with', 'spaces']
```

So it's better to just .split() instead of replacing and splitting by a single whitespace so you don't get so many empty strings in your dict